abstract/suite.go

```diff
@@ -39,6 +40,10 @@ type Suite interface {
 	// Generic constructor to instantiate any abstract interface type
 	// supported by this suite: at least Cipher, Hash, Point, Secret.
 	Constructor
+
+	// NewKey returns a freshly generated private key from the cipher stream.
+	// If cipher == nil, it uses random.Stream.
+	NewKey(cipher.Stream) Secret
 }
```

Comment:
I went for the simplest. We could also have done `NewKeyPair(cipher.Stream) config.KeyPair` but config package already uses abstract, so that would mean to move the `config.KeyPair` into abstract which do not seem something that has to be done in this PR... :)